odxtools/diaglayers/diaglayerraw.py

```diff
@@ -80,88 +80,83 @@ def from_et(et_element: ElementTree.Element, doc_frags: list[OdxDocFragment]) ->
         short_name = odxrequire(et_element.findtext("SHORT-NAME"))
 
         # extend the applicable ODX "document fragments" for the diag layer objects
-        doc_frags = copy(doc_frags)
```

Comment:
Before i undo the change from list to tuple: is there a reason, why this the the only place where a copy of the doc_frags list is created? Can i just reuse the same context.doc_frags as everywhere else?

Reply:
if you make `.doc_frags` a tuple in the `OdxLinkId` and `OdxLinkRef` objects, copying it is not necessary here. in the current version, due to python using references, the `.doc_frag` attributes changing the original list would automagically change all these attributes if no new object was created here...

Reply:
Yes, i understand that, but my question was: is the copying necessary here? The other classes do not copy.

Reply:
I think I only introduced this after something broke down. maybe this simply did not happen for the other classes which introduce a new document fragment. (if this is the case, this is a bug. Possibly it is also the reason for the issues which you've encountered recently.)

I think the list-to-tuple (and maybe freezing the Id and Ref classes) should be factored out into a separate PR and be discussed there, besides this the present PR is mergable IMO...

Reply:
I rebased onto main and removed the tuple change